go-controller/pkg/ovn/pods.go

```diff
@@ -159,44 +159,43 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 		annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
 		if err == nil {
 			podIfAddrs := annotation.IPs
-			_ = oc.lsManager.ReleaseIPs(pod.Spec.NodeName, podIfAddrs)
+			err = oc.lsManager.ReleaseIPs(pod.Spec.NodeName, podIfAddrs)
 		}
-		return
+		return err
 	}
 
 	// FIXME: if any of these steps fails we need to stop and try again later...
 	var allOps, ops []ovsdb.Operation
 	if ops, err = oc.deletePodFromNamespace(pod.Namespace, portInfo); err != nil {
-		klog.Errorf(err.Error())
-	} else {
-		allOps = append(allOps, ops...)
+		return err
 	}
+	allOps = append(allOps, ops...)
 
 	ops, err = oc.ovnNBLSPDel(logicalPort, pod.Spec.NodeName, portInfo.uuid)
 	if err != nil {
-		klog.Errorf(err.Error())
-	} else {
-		allOps = append(allOps, ops...)
+		return err
 	}
+	allOps = append(allOps, ops...)
 
 	_, err = libovsdbops.TransactAndCheck(oc.nbClient, allOps)
 	if err != nil {
-		klog.Errorf("Cannot delete logical switch port %s, %v", logicalPort, err)
+		return fmt.Errorf("cannot delete logical switch port %s, %v", logicalPort, err)
 	}
 
 	if err := oc.lsManager.ReleaseIPs(portInfo.logicalSwitch, portInfo.ips); err != nil {
-		klog.Errorf(err.Error())
+		return err
 	}
 
 	if config.Gateway.DisableSNATMultipleGWs {
 		if err := deletePerPodGRSNAT(oc.nbClient, pod.Spec.NodeName, []*net.IPNet{}, portInfo.ips); err != nil {
-			klog.Errorf(err.Error())
+			return err
 		}
 	}
 	podNsName := ktypes.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}
 	oc.deleteGWRoutesForPod(podNsName, portInfo.ips)
```

Comment:
is it safe for us to continue to assume no errors can come back from this call?

Reply:
yeah good question, maybe I need to convert this function also to return errors?

Reply:
yeah